pdbtools/pdb_selaltloc.py

```diff
@@ -114,338 +120,206 @@ def check_input(args):
+        [A-Z] character.  In these cases, to select the alternative
+        location identified by a blank character give `option=' '`.
 
-    prev_altloc = ''
-    prev_resname = ''
-    prev_resnum = ''
+    Returns
+    -------
+    generator
+        A generator object. To exhaust the generator, that is, to
+        process the PDB file (or PDB lines), convert it to a list.
 
-    # uses the same function names in the loop below. However, depending
-    # on the input options, the functions used are different. One is
-    # specific for byocc=True, and other specific for occ char selection
-    flush_func_multi_residues = flush_resloc_occ if byocc else flush_resloc
+        >>> from pdbtools.pdb_selaltloc import run
+        >>> with('input.pdb', 'r') as fin:
+        >>>     processed_lines = list(run(fin))
 
-    flush_func_single_residues = \
-        flush_resloc_occ_same_residue if byocc else flush_resloc_id_same_residue
+        For more example see:
 
-    # defines records and terminators
+        >>> import pdbtools
+        >>> help(pdbtools)
+    """
     records = ('ATOM', 'HETATM', 'ANISOU')
-    terminators = ('TER', 'END', 'CONECT', 'END', 'ENDMDL')
+    terminators = ('TER', 'END', 'CONECT', 'END', 'ENDMDL', 'MODEL')
+
+    # register atom information
+    register = dict()
+
+    # register comment lines
+    others = []
+
+    # register current chain
+    chain = None
+    prev_chain = None
 
+    # keep record of the line number. This will be used to sort lines
+    # after selecting the desired alternative location
+    nline = 0
+
+    # the loop will collect information on the different atoms
+    # throughout the PDB file until a new chain or any terminal line is
+    # found. At that point, the collected information is flushed because
+    # all altlocs for that block have been defined.
     for line in fhandle:
+        nline += 1
 
         if line.startswith(records):
-            # captures the relevant parameters
-            altloc = line[16]
-            resname = line[17:20]
-            resnum = line[22:26].strip()
-
-            if is_another_altloc_group(
-                    altloc, prev_altloc, resnum, prev_resnum,
-                    resname, prev_resname, altloc_lines, res_per_loc):
-                # if we see the altloc group has changed, we should flush
-                # the lines observed for the previous altloc group
-
-                # uses "for loop" instead of "yield from" to maintain
-                # compatibility with older python version
-                if partial_altloc(altloc_lines):
-                    flush_func = flush_func_single_residues
-                else:
-                    flush_func = flush_func_multi_residues
 
-                for __line in flush_func(selloc=selloc, altloc_lines=altloc_lines):
-                    yield __line
+            # here resnum + insertion code are taken to identify
+            # different residues
+            resnum = line[22:27].strip()
```

Comment:
You don't need to strip the resnum if you're only using it as an identifier. Saves a few cycles..